### PR TITLE
[ci/release/ml] Run ML release tests on staging

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

This moves all ML release tests to staging. I will run smoke tests before merging.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by krfricke** [comment]
Build: https://buildkite.com/ray-project/release-tests-branch/builds/725

**Comment by krfricke** [comment]
Next try: https://buildkite.com/ray-project/release-tests-branch/builds/731

File: release/release_tests.yaml [hunk 1]
```diff
@@ -2579,6 +2638,7 @@
   team: core
   frequency: nightly
   working_dir: microbenchmark
+  env: prod
```
Comment (rkooo567) [inline release/release_tests.yaml:2641]: is it a mistake ?

**Review by rkooo567 — APPROVED** [review]

**Comment by krfricke** [comment]
K8s passing, lightning still failing. Next build: https://buildkite.com/ray-project/release-tests-branch/builds/735

File: release/release_tests.yaml (hunk 1 above)
Comment (krfricke) [reply release/release_tests.yaml:2641]: No this is deliberate - there is a `microbenchmark_staging` test and `microbenchmark` should be on prod. This is to make sure this remains after we eventually make a switch to default to staging

**Pull request merged**